src/System.Private.CoreLib/shared/System/Buffers/Utilities.cs

```diff
@@ -14,6 +15,12 @@ internal static int SelectBucketIndex(int bufferSize)
         {
             Debug.Assert(bufferSize >= 0);
 
+            if (Lzcnt.IsSupported && IntPtr.Size == 8)
```

Comment:
This works on 32-bit as well, no need to test `IntPtr.Size` here.

Reply:
@pentp are you sure? I had to add that check to fix few failing CI agents. Like here https://github.com/dotnet/coreclr/blob/master/tests/src/JIT/HardwareIntrinsics/X86/Lzcnt/Lzcnt.cs#L52

Reply:
64-bit test is needed for `Lzcnt(long)` overload, the `int` overload works on both.

Reply:
ah, makes sense, thanks